pyfar/utils.py

```diff
@@ -0,0 +1,129 @@
+import pyfar as pf
+import numpy as np
+
+
+def broadcast_cshape(signal, cshape):
+    """
+    Broadcast signal to cshape.
+
+    The :py:mod:`cshape <pyfar._concepts.audio_classes>` of the signal is
+    broadcasted following the `numpy broadcasting rules
+    <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+
+    Parameters
+    ----------
+    signal : Signal, TimeData, FrequencyData
+        The signal to be broadcasted.
+    cshape : tuple
+        The cshape to which `signal` is broadcasted.
+
+    Returns
+    -------
+    signal : Signal, TimeData, FrequencyData
+        The broadcasted input signal
```

Comment:
Should we mention explicitly that (in contrast to numpy) a copy of the signal is returned?

Reply:
Yes, good idea

Reply:
Could also be added to the other functions :)

Reply:
True, added to all functions